### DISCUSSION TITLE
How to get raw-log URL from downloadJobLogsForWorkflowRun without downloading

### DISCUSSION THREAD

**Discussion by jeroen** [octokit/octokit.js#1805] [General]
I would like to show the raw log file for a job in the browser. Simplified:

```js
octokit.actions.downloadJobLogsForWorkflowRun({
  owner : "r-devel",
  repo: "r-svn",
  job_id: job.id
}).then(function(x){
  window.open(x.url);
});
```

This works, but it is not ideal because it doesn't show the popup until after the entire log file is downloaded.

This is because the call to `https://api.github.com/repos/owner/repo/actions/jobs/{{id}}/logs` redirects with a 302 to the final URL of the log file, which takes a long time to download.

Instead I would like to just get the temporary raw-log url, and open that url in a new window, without first waiting for octo.js to follow the original redirect to download the entire raw log file in ajax. Is this possible?

**Comment by gr2m** [comment]
Try adding `method: "HEAD"`:

```js
octokit.actions.downloadJobLogsForWorkflowRun({
  method: "HEAD",
  owner : "r-devel",
  repo: "r-svn",
  job_id: job.id
}).then(function(x){
  window.open(x.url);
});
```

That should follow the redirect, without downloading the final response body

**Reply by jeroen** [reply]
This gives me a CORS error :(

```
Access to fetch at 'https://pipelines.actions.githubusercontent.com/.......................' (redirected from 'https://api.github.com/repos/r-devel/r-svn/actions/jobs/890258197/logs') from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Reply by gr2m** [reply]
odd that you don't get the CORS error when not using `HEAD`. Are you running the code in a browser, or a different JavaScript runtime environment such as Node or Electron?

I wish there was a way to just receive the final download URL. I've been asking for it for a long time now, it's a problem for browser apps. I will add your use case to https://github.com/gr2m/github-api-wishlist/tree/master/wishlist/link-media-type.

**Reply by jeroen** [reply]
I'm just using google chrome.

**Reply by gr2m** [reply]
Unfortunately there is not much we can do. The final redirect URL is not exposed on any endpoint on api.github.com, that's what I'm asking for at https://github.com/gr2m/github-api-wishlist/tree/master/wishlist/link-media-type. 

The other thing I keep asking for is the same CORS settings for adjacent domains, which would fix the CORS error you are seeing.
https://github.com/gr2m/github-api-wishlist/tree/master/wishlist/cors-for-adjacent-domains

All you can do at this point is to contact support and bring the problem to their attention, to bump the priority: https://support.github.com/contact

I'm sorry that there is no solution at this point for browser users

**Reply by gr2m** [reply]
In this particular case, the domain is https://pipelines.actions.githubusercontent.com. CORS headers are actually set for this one, but the problem is the `access-control-allow-methods` response header, which is set to `OPTIONS,GET,POST,PATCH,PUT,DELETE`

If it would include `HEAD`, then the above code would work. Could you please ask support about this? They will forward your question to the right team.

Minimal test case to reproduce the problem

This works

```js
fetch('https://api.github.com/repos/[OWNER]/[REPO]/actions/jobs/[JOB ID]/logs', {
  headers: { authorization: 'token [TOKEN]'}
})
```

This does not

```js
fetch('https://api.github.com/repos/[OWNER]/[REPO]/actions/jobs/[JOB ID]/logs', {
  method: "HEAD",
  headers: { authorization: 'token [TOKEN]'}
})
```
